w3lib/url.py

```diff
@@ -374,7 +374,15 @@ def _safe_ParseResult(parts, encoding='utf8', path_encoding='utf8'):
     # IDNA encoding can fail for too long labels (>63 characters)
     # or missing labels (e.g. http://.example.com)
     try:
-        netloc = parts.netloc.encode('idna')
+        idx = parts.netloc.rfind(u':')
```

Comment:
What do you think of using `.hostname` and `.port` attributes from the `SplitResult` instead of looking for a port number in netloc?